assets/src/dashboard/utils/usePagePreviewSize.js

```diff
@@ -98,18 +104,32 @@ export default function usePagePreviewSize(options = {}) {
       return () => {};
     }
 
-    const handleResize = () => debounceAvailableContainerSpace();
-
-    window.addEventListener('resize', handleResize);
+    window.addEventListener('resize', debounceAvailableContainerSpace);
     return () => {
-      window.removeEventListener('resize', handleResize);
+      window.removeEventListener('resize', debounceAvailableContainerSpace);
     };
   }, [thumbnailMode, debounceAvailableContainerSpace]);
 
   useEffect(() => setBp(getCurrentBp(availableContainerSpace)), [
     availableContainerSpace,
   ]);
 
+  useEffect(() => setAvailableContainerSpace(getTrueInnerWidth(wpMenuWidth)), [
+    wpMenuWidth,
+  ]);
+
+  useEffect(() => {
+    if (thumbnailMode || !wpLeftNav.current) {
+      return () => {};
+    }
+    const wpLeftNavCurrent = wpLeftNav.current;
+
+    wpLeftNavCurrent.addEventListener('click', debounceWpMenuToggle);
```

Comment:
I don't like how tightly coupled this code is to WordPress. Especially as it relies on detecting clicks on this toggle button. It'd be more reliable to check for the `folded` class on the `<body>`. However, it is still coupled to WordPress.

We went down that road for the editor already and ended up implementing a [`useResizeEffect`](https://github.com/google/web-stories-wp/blob/200897d5d911bf6aa20faf777b293a8f6d28f78f/assets/src/edit-story/utils/useResizeEffect.js) hook to detect how much space is available. Here's how we currently use it in the editor's carousel:

https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/components/canvas/carousel/index.js#L221-L231

Bonus: if you adopt this hook in the dashboard, you could also use it in `ToggleButtonGroup`.

**tl;dr**: let's avoid relying on WP environment specifics.

Reply:
thanks for this note, @swissspidy! I refactored to use the approach you suggested - it was something we were hesitant to spend the time on but i appreciate the desire for continuity. I think it turned out nice if you wanna take another peek. I did update the useResizeEffect slightly to check that the ref exists. before grabbing .current and in cases where that node is false returning undefined to remove that console warning about not returning null.